[java] NPE applying rule JUnitTestsShouldIncludeAssert

**Rule Set:**
Rule set file can be found on this [page](https://github.com/kasirgalabs/ETUmulator/blob/20051d02a854e5327de36a4f397778c1ab1d4d4e/config/pmd/kasirgalabs-pmd.xml).
**Description:**
Every time when I run PMD on the given code NullPointerException is thrown when processing rule `JUnitTestsShouldIncludeAssert`.
**Code Sample demonstrating the issue:**
Code that PMD trying to process can be found on this [page](https://github.com/kasirgalabs/ETUmulator/blob/20051d02a854e5327de36a4f397778c1ab1d4d4e/src/test/java/com/kasirgalabs/etumulator/console/BaseConsoleTest.java).
Travic CI build log can be found on this [page](https://travis-ci.org/kasirgalabs/ETUmulator/builds/219338890).
**Running PMD through:**
Gradle

Travis CI build log contains the thrown exception but I'll also write it here.
```
Exception applying rule JUnitTestsShouldIncludeAssert on file /home/travis/build/kasirgalabs/ETUmulator/src/test/java/com/kasirgalabs/etumulator/console/BaseConsoleTest.java, continuing with next rule
java.lang.NullPointerException
	at net.sourceforge.pmd.lang.java.rule.junit.JUnitTestsShouldIncludeAssertRule.isExpectStatement(JUnitTestsShouldIncludeAssertRule.java:144)
	at net.sourceforge.pmd.lang.java.rule.junit.JUnitTestsShouldIncludeAssertRule.containsExpectOrAssert(JUnitTestsShouldIncludeAssertRule.java:54)
	at net.sourceforge.pmd.lang.java.rule.junit.JUnitTestsShouldIncludeAssertRule.containsExpectOrAssert(JUnitTestsShouldIncludeAssertRule.java:61)
	at net.sourceforge.pmd.lang.java.rule.junit.JUnitTestsShouldIncludeAssertRule.containsExpectOrAssert(JUnitTestsShouldIncludeAssertRule.java:61)
	at net.sourceforge.pmd.lang.java.rule.junit.JUnitTestsShouldIncludeAssertRule.containsExpectOrAssert(JUnitTestsShouldIncludeAssertRule.java:61)
	at net.sourceforge.pmd.lang.java.rule.junit.JUnitTestsShouldIncludeAssertRule.visit(JUnitTestsShouldIncludeAssertRule.java:44)
	at net.sourceforge.pmd.lang.java.ast.ASTMethodDeclaration.jjtAccept(ASTMethodDeclaration.java:25)
	at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.childrenAccept(AbstractJavaNode.java:55)
	at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.visit(AbstractJavaRule.java:74)
	at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.visit(AbstractJavaRule.java:111)
	at net.sourceforge.pmd.lang.java.ast.ASTClassOrInterfaceBodyDeclaration.jjtAccept(ASTClassOrInterfaceBodyDeclaration.java:41)
	at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.childrenAccept(AbstractJavaNode.java:55)
	at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.visit(AbstractJavaRule.java:74)
	at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.visit(AbstractJavaRule.java:107)
	at net.sourceforge.pmd.lang.java.ast.ASTClassOrInterfaceBody.jjtAccept(ASTClassOrInterfaceBody.java:22)
	at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.childrenAccept(AbstractJavaNode.java:55)
	at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.visit(AbstractJavaRule.java:74)
	at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.visit(AbstractJavaRule.java:83)
	at net.sourceforge.pmd.lang.java.rule.junit.JUnitTestsShouldIncludeAssertRule.visit(JUnitTestsShouldIncludeAssertRule.java:34)
	at net.sourceforge.pmd.lang.java.ast.ASTClassOrInterfaceDeclaration.jjtAccept(ASTClassOrInterfaceDeclaration.java:30)
	at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.childrenAccept(AbstractJavaNode.java:55)
	at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.visit(AbstractJavaRule.java:74)
	at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.visit(AbstractJavaRule.java:223)
	at net.sourceforge.pmd.lang.java.ast.ASTTypeDeclaration.jjtAccept(ASTTypeDeclaration.java:36)
	at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.childrenAccept(AbstractJavaNode.java:55)
	at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.visit(AbstractJavaRule.java:74)
	at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.visit(AbstractJavaRule.java:203)
	at net.sourceforge.pmd.lang.java.rule.junit.AbstractJUnitRule.visit(AbstractJUnitRule.java:64)
	at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.visitAll(AbstractJavaRule.java:31)
	at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.apply(AbstractJavaRule.java:25)
	at net.sourceforge.pmd.lang.rule.AbstractDelegateRule.apply(AbstractDelegateRule.java:215)
	at net.sourceforge.pmd.RuleSet.apply(RuleSet.java:306)
	at net.sourceforge.pmd.RuleSets.apply(RuleSets.java:125)
	at net.sourceforge.pmd.SourceCodeProcessor.processSource(SourceCodeProcessor.java:150)
	at net.sourceforge.pmd.SourceCodeProcessor.processSourceCode(SourceCodeProcessor.java:81)
	at net.sourceforge.pmd.SourceCodeProcessor.processSourceCode(SourceCodeProcessor.java:47)
	at net.sourceforge.pmd.processor.PmdRunnable.call(PmdRunnable.java:87)
	at net.sourceforge.pmd.processor.MonoThreadProcessor.runAnalysis(MonoThreadProcessor.java:28)
	at net.sourceforge.pmd.processor.AbstractPMDProcessor.processFiles(AbstractPMDProcessor.java:74)
	at net.sourceforge.pmd.PMD.processFiles(PMD.java:376)
	at net.sourceforge.pmd.ant.internal.PMDTaskImpl.doTask(PMDTaskImpl.java:187)
	at net.sourceforge.pmd.ant.internal.PMDTaskImpl.execute(PMDTaskImpl.java:268)
	at net.sourceforge.pmd.ant.PMDTask.execute(PMDTask.java:47)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:292)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
	at groovy.util.AntBuilder.performTask(AntBuilder.java:327)
	at groovy.util.AntBuilder.nodeCompleted(AntBuilder.java:272)
	at org.gradle.api.internal.project.ant.BasicAntBuilder.nodeCompleted(BasicAntBuilder.java:78)
	at sun.reflect.GeneratedMethodAccessor43.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite$PojoCachedMethodSiteNoUnwrapNoCoerce.invoke(PojoMetaMethodSite.java:233)
	at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite.call(PojoMetaMethodSite.java:56)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
	at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite.call(PojoMetaMethodSite.java:58)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:133)
	at org.gradle.api.internal.project.antbuilder.AntBuilderDelegate.nodeCompleted(AntBuilderDelegate.groovy:80)
	at groovy.util.BuilderSupport.doInvokeMethod(BuilderSupport.java:149)
	at groovy.util.BuilderSupport.invokeMethod(BuilderSupport.java:66)
	at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:48)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:133)
	at org.gradle.api.plugins.quality.Pmd$_run_closure2.doCall(Pmd.groovy:168)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
	at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:294)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1019)
	at groovy.lang.Closure.call(Closure.java:426)
	at groovy.lang.Closure.call(Closure.java:442)
	at org.gradle.api.internal.ClosureBackedAction.execute(ClosureBackedAction.java:67)
	at org.gradle.util.ConfigureUtil.configure(ConfigureUtil.java:130)
	at org.gradle.util.ConfigureUtil.configure(ConfigureUtil.java:91)
	at org.gradle.api.internal.project.antbuilder.DefaultIsolatedAntBuilder$2.execute(DefaultIsolatedAntBuilder.java:140)
	at org.gradle.api.internal.project.antbuilder.DefaultIsolatedAntBuilder$2.execute(DefaultIsolatedAntBuilder.java:122)
	at org.gradle.api.internal.project.antbuilder.ClassPathToClassLoaderCache.withCachedClassLoader(ClassPathToClassLoaderCache.java:134)
	at org.gradle.api.internal.project.antbuilder.DefaultIsolatedAntBuilder.execute(DefaultIsolatedAntBuilder.java:116)
	at org.gradle.api.internal.project.IsolatedAntBuilder$execute$0.call(Unknown Source)
	at org.gradle.api.plugins.quality.Pmd.run(Pmd.groovy:166)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:75)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.doExecute(AnnotationProcessingTaskFactory.java:227)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.execute(AnnotationProcessingTaskFactory.java:220)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.execute(AnnotationProcessingTaskFactory.java:209)
	at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:585)
	at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:568)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:80)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:61)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:46)
	at org.gradle.api.internal.tasks.execution.PostExecutionAnalysisTaskExecuter.execute(PostExecutionAnalysisTaskExecuter.java:35)
	at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:64)
	at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
	at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:52)
	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:53)
	at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:203)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:185)
	at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.processTask(AbstractTaskPlanExecutor.java:66)
	at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.run(AbstractTaskPlanExecutor.java:50)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor.process(DefaultTaskPlanExecutor.java:25)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter.execute(DefaultTaskGraphExecuter.java:110)
	at org.gradle.execution.SelectedTaskExecutionAction.execute(SelectedTaskExecutionAction.java:37)
	at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:37)
	at org.gradle.execution.DefaultBuildExecuter.access$000(DefaultBuildExecuter.java:23)
	at org.gradle.execution.DefaultBuildExecuter$1.proceed(DefaultBuildExecuter.java:43)
	at org.gradle.execution.DryRunBuildExecutionAction.execute(DryRunBuildExecutionAction.java:32)
	at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:37)
	at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:30)
	at org.gradle.initialization.DefaultGradleLauncher$4.run(DefaultGradleLauncher.java:154)
	at org.gradle.internal.Factories$1.create(Factories.java:22)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:52)
	at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:151)
	at org.gradle.initialization.DefaultGradleLauncher.access$200(DefaultGradleLauncher.java:32)
	at org.gradle.initialization.DefaultGradleLauncher$1.create(DefaultGradleLauncher.java:99)
	at org.gradle.initialization.DefaultGradleLauncher$1.create(DefaultGradleLauncher.java:93)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:62)
	at org.gradle.initialization.DefaultGradleLauncher.doBuild(DefaultGradleLauncher.java:93)
	at org.gradle.initialization.DefaultGradleLauncher.run(DefaultGradleLauncher.java:82)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter$DefaultBuildController.run(InProcessBuildActionExecuter.java:94)
	at org.gradle.tooling.internal.provider.ExecuteBuildActionRunner.run(ExecuteBuildActionRunner.java:28)
	at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:43)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:28)
	at org.gradle.launcher.exec.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:78)
	at org.gradle.launcher.exec.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:48)
	at org.gradle.launcher.exec.DaemonUsageSuggestingBuildActionExecuter.execute(DaemonUsageSuggestingBuildActionExecuter.java:51)
	at org.gradle.launcher.exec.DaemonUsageSuggestingBuildActionExecuter.execute(DaemonUsageSuggestingBuildActionExecuter.java:28)
	at org.gradle.launcher.cli.RunBuildAction.run(RunBuildAction.java:43)
	at org.gradle.internal.Actions$RunnableActionAdapter.execute(Actions.java:170)
	at org.gradle.launcher.cli.CommandLineActionFactory$ParseAndBuildAction.execute(CommandLineActionFactory.java:237)
	at org.gradle.launcher.cli.CommandLineActionFactory$ParseAndBuildAction.execute(CommandLineActionFactory.java:210)
	at org.gradle.launcher.cli.JavaRuntimeValidationAction.execute(JavaRuntimeValidationAction.java:35)
	at org.gradle.launcher.cli.JavaRuntimeValidationAction.execute(JavaRuntimeValidationAction.java:24)
	at org.gradle.launcher.cli.CommandLineActionFactory$WithLogging.execute(CommandLineActionFactory.java:206)
	at org.gradle.launcher.cli.CommandLineActionFactory$WithLogging.execute(CommandLineActionFactory.java:169)
	at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:33)
	at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:22)
	at org.gradle.launcher.Main.doAction(Main.java:33)
	at org.gradle.launcher.bootstrap.EntryPoint.run(EntryPoint.java:45)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.launcher.bootstrap.ProcessBootstrap.runNoExit(ProcessBootstrap.java:54)
	at org.gradle.launcher.bootstrap.ProcessBootstrap.run(ProcessBootstrap.java:35)
	at org.gradle.launcher.GradleMain.main(GradleMain.java:23)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.wrapper.BootstrapMainStarter.start(BootstrapMainStarter.java:30)
	at org.gradle.wrapper.WrapperExecutor.execute(WrapperExecutor.java:129)
	at org.gradle.wrapper.GradleWrapperMain.main(GradleWrapperMain.java:61)
```

Similar NPE:
```
Caused by: java.lang.NullPointerException
	at net.sourceforge.pmd.lang.ast.AbstractNode.jjtGetChild(AbstractNode.java:103)
	at net.sourceforge.pmd.lang.java.rule.junit.JUnitTestsShouldIncludeAssertRule.isExpectStatement(JUnitTestsShouldIncludeAssertRule.java:144)
	at net.sourceforge.pmd.lang.java.rule.junit.JUnitTestsShouldIncludeAssertRule.containsExpectOrAssert(JUnitTestsShouldIncludeAssertRule.java:54)
	at net.sourceforge.pmd.lang.java.rule.junit.JUnitTestsShouldIncludeAssertRule.containsExpectOrAssert(JUnitTestsShouldIncludeAssertRule.java:61)
	at net.sourceforge.pmd.lang.java.rule.junit.JUnitTestsShouldIncludeAssertRule.containsExpectOrAssert(JUnitTestsShouldIncludeAssertRule.java:61)
	at net.sourceforge.pmd.lang.java.rule.junit.JUnitTestsShouldIncludeAssertRule.containsExpectOrAssert(JUnitTestsShouldIncludeAssertRule.java:61)
	at net.sourceforge.pmd.lang.java.rule.junit.JUnitTestsShouldIncludeAssertRule.containsExpectOrAssert(JUnitTestsShouldIncludeAssertRule.java:61)
	at net.sourceforge.pmd.lang.java.rule.junit.JUnitTestsShouldIncludeAssertRule.containsExpectOrAssert(JUnitTestsShouldIncludeAssertRule.java:61)
	at net.sourceforge.pmd.lang.java.rule.junit.JUnitTestsShouldIncludeAssertRule.containsExpectOrAssert(JUnitTestsShouldIncludeAssertRule.java:61)
	at net.sourceforge.pmd.lang.java.rule.junit.JUnitTestsShouldIncludeAssertRule.containsExpectOrAssert(JUnitTestsShouldIncludeAssertRule.java:61)
	at net.sourceforge.pmd.lang.java.rule.junit.JUnitTestsShouldIncludeAssertRule.containsExpectOrAssert(JUnitTestsShouldIncludeAssertRule.java:61)
	at net.sourceforge.pmd.lang.java.rule.junit.JUnitTestsShouldIncludeAssertRule.containsExpectOrAssert(JUnitTestsShouldIncludeAssertRule.java:61)
	at net.sourceforge.pmd.lang.java.rule.junit.JUnitTestsShouldIncludeAssertRule.containsExpectOrAssert(JUnitTestsShouldIncludeAssertRule.java:61)
	at net.sourceforge.pmd.lang.java.rule.junit.JUnitTestsShouldIncludeAssertRule.containsExpectOrAssert(JUnitTestsShouldIncludeAssertRule.java:61)
	at net.sourceforge.pmd.lang.java.rule.junit.JUnitTestsShouldIncludeAssertRule.containsExpectOrAssert(JUnitTestsShouldIncludeAssertRule.java:61)
	at net.sourceforge.pmd.lang.java.rule.junit.JUnitTestsShouldIncludeAssertRule.containsExpectOrAssert(JUnitTestsShouldIncludeAssertRule.java:61)
	at net.sourceforge.pmd.lang.java.rule.junit.JUnitTestsShouldIncludeAssertRule.containsExpectOrAssert(JUnitTestsShouldIncludeAssertRule.java:61)
	at net.sourceforge.pmd.lang.java.rule.junit.JUnitTestsShouldIncludeAssertRule.containsExpectOrAssert(JUnitTestsShouldIncludeAssertRule.java:61)
	at net.sourceforge.pmd.lang.java.rule.junit.JUnitTestsShouldIncludeAssertRule.visit(JUnitTestsShouldIncludeAssertRule.java:44)
	at net.sourceforge.pmd.lang.java.ast.ASTMethodDeclaration.jjtAccept(ASTMethodDeclaration.java:25)
	at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.childrenAccept(AbstractJavaNode.java:56)
	at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.visit(AbstractJavaRule.java:75)
	at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.visit(AbstractJavaRule.java:112)
	at net.sourceforge.pmd.lang.java.ast.ASTClassOrInterfaceBodyDeclaration.jjtAccept(ASTClassOrInterfaceBodyDeclaration.java:41)
	at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.childrenAccept(AbstractJavaNode.java:56)
	at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.visit(AbstractJavaRule.java:75)
	at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.visit(AbstractJavaRule.java:108)
	at net.sourceforge.pmd.lang.java.ast.ASTClassOrInterfaceBody.jjtAccept(ASTClassOrInterfaceBody.java:21)
	at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.childrenAccept(AbstractJavaNode.java:56)
	at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.visit(AbstractJavaRule.java:75)
	at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.visit(AbstractJavaRule.java:84)
	at net.sourceforge.pmd.lang.java.rule.junit.JUnitTestsShouldIncludeAssertRule.visit(JUnitTestsShouldIncludeAssertRule.java:34)
	at net.sourceforge.pmd.lang.java.ast.ASTClassOrInterfaceDeclaration.jjtAccept(ASTClassOrInterfaceDeclaration.java:30)
	at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.childrenAccept(AbstractJavaNode.java:56)
	at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.visit(AbstractJavaRule.java:75)
	at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.visit(AbstractJavaRule.java:224)
	at net.sourceforge.pmd.lang.java.ast.ASTTypeDeclaration.jjtAccept(ASTTypeDeclaration.java:35)
	at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.childrenAccept(AbstractJavaNode.java:56)
	at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.visit(AbstractJavaRule.java:75)
	at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.visit(AbstractJavaRule.java:204)
	at net.sourceforge.pmd.lang.java.rule.junit.AbstractJUnitRule.visit(AbstractJUnitRule.java:66)
	at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.visitAll(AbstractJavaRule.java:32)
	at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.apply(AbstractJavaRule.java:26)
	at net.sourceforge.pmd.RuleSet.apply(RuleSet.java:453)
```

With this test case:
```
    <test-code>
        <description>Anonymous class initializer accessing member field should work</description>
        <expected-problems>0</expected-problems>
        <code><![CDATA[
public class Foo {
    @Test
    public void testName() {
        Foo foo = new Foo() {
            {
                this.field = null;
            }
        };
    }
}
        ]]></code>
    </test-code>
```

--- a/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/junit/JUnitTestsShouldIncludeAssertRule.java
+++ b/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/junit/JUnitTestsShouldIncludeAssertRule.java
@@ -124,10 +124,13 @@ public class JUnitTestsShouldIncludeAssertRule extends AbstractJUnitRule {
         if (expression != null) {
             ASTPrimaryExpression pe = expression.getFirstChildOfType(ASTPrimaryExpression.class);
             if (pe != null) {
-                String img = pe.jjtGetChild(0).jjtGetChild(0).getImage();
-                if (img != null && (img.startsWith("assert") || img.startsWith("fail")
-                        || img.startsWith("Assert.assert") || img.startsWith("Assert.fail"))) {
-                    return true;
+                Node name = pe.getFirstDescendantOfType(ASTName.class);
+                if (name != null) {
+                    String img = name.getImage();
+                    if (img != null && (img.startsWith("assert") || img.startsWith("fail")
+                            || img.startsWith("Assert.assert") || img.startsWith("Assert.fail"))) {
+                        return true;
+                    }
                 }
             }
         }
@@ -141,14 +144,13 @@ public class JUnitTestsShouldIncludeAssertRule extends AbstractJUnitRule {
             
             ASTPrimaryExpression pe = expression.getFirstChildOfType(ASTPrimaryExpression.class);
             if (pe != null) {
-                Node subChild = pe.jjtGetChild(0).jjtGetChild(0);
-                
-                // case of eg AllocationExpression
-                if (!(subChild instanceof ASTName)) {
+                Node name = pe.getFirstDescendantOfType(ASTName.class);
+                // case of an AllocationExpression
+                if (name == null) {
                     return false;
                 }
                 
-                String img = subChild.getImage();
+                String img = name.getImage();
                 if (img.indexOf(".") == -1) {
                     return false;
                 }
@@ -159,7 +161,7 @@ public class JUnitTestsShouldIncludeAssertRule extends AbstractJUnitRule {
                 }
 
                 for (NameOccurrence occ : expectables.get(varname)) {
-                    if (occ.getLocation() == pe.jjtGetChild(0).jjtGetChild(0) && img.startsWith(varname + ".expect")) {
+                    if (occ.getLocation() == name && img.startsWith(varname + ".expect")) {
                         return true;
                     }
                 }

--- a/pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/junit/xml/JUnitTestsShouldIncludeAssert.xml
+++ b/pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/junit/xml/JUnitTestsShouldIncludeAssert.xml
@@ -328,5 +328,16 @@ public class BaseConsoleTest extends UART {
 }
         ]]></code>
     </test-code>
-
+    <test-code>
+        <description>#330 NPE accessing local field / method with this.XXX</description>
+        <expected-problems>1</expected-problems>
+        <code><![CDATA[
+public class Foo {
+    @Test
+    public void testName() {
+        this.field = null;
+    }
+}
+        ]]></code>
+    </test-code>
 </test-data>

